### ISSUE TITLE
viewable HTML, PDF  for Lisp source code and documentation 

### ISSUE THREAD

**Issue by masinter** [Interlisp/medley#966]
It seems like there's interest in lsee and seetops but these won't work well for files expecting PRETTYFILEINEX.  So I was thinking of generating _something_ printable and viewable for all lisp sources and tedit files in the repo and making them available automatically -- whenever there's a push?

maybe lsee-able (PRETTYFILEINDEX), HTML (viewable), PDF.?

Generating PDF would entail PRETTYFILEINDEX -> seetops -> ps2pdf for lisp sources, and HCFILES (in test repo).
Generating HTML would require a TEdit converter (not quite an image stream) and a different filter for  seetops.

This issue is to gather ideas of what you think people would find useful.

**Comment by masinter** [comment]
The link https://drive.google.com/drive/folders/15_lL8F80xEVsJi3S9iyDLKW4_Ujcqelr?usp=sharing
will give you a list of (almost all) Interlisp source files prettyprinted with "PRETTYFILEINDEX", written to Postscript and converted to PDF.

This could get better in several ways

- [ ] Group the files into logical units and use MULTIFILEINDEX
- [ ] automate the process when files are written (there's code to do that)
- [ ] there are some files that have package problems -- it just skipped them
- [ ] pick better fonts

but this is at least a "proof of concept". We could do MAKEFILE(... (FAST NEW)) and not worry about font changes any more.

**Comment by rmkaplan** [comment]
I agree, we don’t need to do the font changes in the files if we are willing to reformat the files any time we want to look at them, which is what prettyfileindex does.  But then you don’t actually see what’s in the file, you see it’s theory of what’s in the file, not always correct.

TEDIT-SEE either respects the font changes in the file, or calls PRETTYFILEINDEX to print the file to a tedit stream.

Similarly, PF (or TF, for Tedit-function) also does it either way.

Although it may be nice to have PS and PDF files to view a whole file outside of Medley, I don’t think that addresses the functionality of wanting to have a scrollable version of a file or function visible (and selectable) from inside Medley.

At least for viewing, we would also have to solve the problem (from my previous request) of making it easy to call up a ps/pdf file from the SEE or TEDIT-SEE commands.

**Comment by masinter** [comment]
@rmkaplan These PDF files aren't meant for you. They're for people who don't want to load or run anything, they just want to SEE the source code without running anything else. 
PDF is OK, HTML might be better, but this was quick (it took a few hours -- in the file test/new/printing/HCFILES

**Comment by rmkaplan** [comment]
OK, thanks.

**Comment by masinter** [comment]
PDF files under medley-sources/out in [this Google Drive folder](https://drive.google.com/drive/folders/10ZBQty5gEwdBnZHtEbXfe5f1dHGziGZG?usp=sharing)? 

I renamed the folder to be [medley repo source listings (in PDF)](https://drive.google.com/open?id=15Wl8bNAobimtH0iOQsQ38nMwFcjoPzWB&usp=drive_copy) to be a little more explanatory but I think the source listings
1) should be easier to find
2) should be generated automatically (nightly? )

on the website? How indexed?

- [ ]  Group the files into logical units and use MULTIFILEINDEX
- [ ]  automate the process when files are written
- [ ]  there are some files that have package problems -- it just skipped them
 better fonts

- [ ] I hve a personal fontset which only the display font sizes.

**Comment by masinter** [comment]
Starting with https://interlisp.org/medley/index.html
you can have access on the web to any file in the Medley repository with Lisp source files 're-pretty-printed'- TEdit and other formats TEdit can read also converted to PDF.
The implementation of the feature is still preliminary but I'm hoping this will be helpful.

The implementation so far is in the internal/MEDLEY-UTILS file, under the name HCFILES and PRETTYFILES for generating the PDFs, and MAKE-FILES-INDEX for building a "index.html" for every directory.

Both the PDFs and the index.html should only be pushed into the "Gh-pages" repo whenever a new version of a .TEDIT or lisp sourcefile has been pushed to Master. Some things yet to do:

- [ ] make index pages prettier (group by root and then by source / translation)
- [ ] automatically derive a PSEUDOHOST filesystem for remote access to the latest Master... right now the automation is to rebuild everything from scratch. 
- [ ] automate a process for selectively re-printing only the files where the content changed..
- [ ] 

- [ ]

**Comment by rmkaplan** [comment]
This is great!

One think I see is that there are a lot of links to the actual sources files (not the pdf’s) that go to 404.  Are you planning to include the actual loadable files here, or just their rendered images?
